cookbooks/agents_with_memory.ipynb

```diff
@@ -0,0 +1,728 @@
+      ],
+      "metadata": {
+        "id": "Cg96MkbcRtQR"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "collapsed": true,
+        "id": "ZVmDAK6MPefC"
+      },
+      "outputs": [],
+      "source": [
+        "pip install camel-ai[all]==0.1.6.8"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### 🔑 Setting Up API Keys"
+      ],
+      "metadata": {
+        "id": "MyTTCe3IR_Lr"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "You'll need to set up your API keys for OpenAI."
+      ],
+      "metadata": {
+        "id": "REqzgGL9SEaD"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import os\n",
+        "from getpass import getpass\n",
+        "\n",
+        "# Prompt for the API key securely\n",
+        "openai_api_key = getpass('Enter your API key: ')\n",
+        "os.environ[\"OPENAI_API_KEY\"] = openai_api_key"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "PNBFEXc-R-0s",
+        "outputId": "52d481de-9a81-4c00-e3db-9c468e6b5b22"
+      },
+      "execution_count": 2,
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Enter your API key: ··········\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Usage"
+      ],
+      "metadata": {
+        "id": "LRojeqp7dP1m"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "To use the Memory module in your agent:\n",
+        "\n",
+        "1. Choose an appropriate AgentMemory implementation (`ChatHistoryMemory`, `VectorDBMemory`, or `LongtermAgentMemory`).\n",
+        "2. Initialize the memory with a context creator and any necessary parameters.\n",
+        "3. Use `write_records()` to add new information to the memory.\n",
+        "4. Use `retrieve()` to get relevant context for the agent's next action.\n",
+        "5. Use `get_context()` to obtain the formatted context for the agent."
+      ],
+      "metadata": {
+        "id": "_sJV9GFldTBh"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Setting `LongtermAgentMemory`:"
+      ],
+      "metadata": {
+        "id": "aiypnpqJoZ4G"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from camel.memories import (\n",
+        "    ChatHistoryBlock,\n",
+        "    LongtermAgentMemory,\n",
+        "    MemoryRecord,\n",
+        "    ScoreBasedContextCreator,\n",
+        "    VectorDBBlock,\n",
+        ")\n",
+        "from camel.messages import BaseMessage\n",
+        "from camel.types import ModelType, OpenAIBackendRole\n",
+        "from camel.utils import OpenAITokenCounter\n",
+        "\n",
```

Comment:
This part of the code is a bit long, maybe we can move the part of importing modules out, like in other cookbooks.WDYT

Reply:
Good point, thanks